clang/tools/clang-offload-bundler/ClangOffloadBundler.cpp

```diff
@@ -643,9 +643,22 @@ class ObjectFileHandler final : public FileHandler {
             return createStringError(inconvertibleErrorCode(),
                                      Err.getMessage());
 
+          bool UpdateBuf = false;
           if (!Mod->getModuleInlineAsm().empty()) {
             Mod->setModuleInlineAsm("");
-
+            UpdateBuf = true;
+          }
+          for (auto I = Mod->global_begin(), E = Mod->global_end(); I != E;) {
+            GlobalVariable &GV = *I++;
+            // Do not add globals with constant address space to the tgtsym.
+            if (!GV.isDeclaration() && !GV.hasLocalLinkage() &&
+                GV.getAddressSpace() == 2) {
```

Comment:
May we use clang::LangAS enum value here instead of 2?

Reply:
Do you mean clang::LangAS::opencl_constant? It has value 3.

Reply:
I'm just wondering where constant AS is set to 2 instead of 3.

Reply:
AFAIK clang AST address spaces (`clang::LangAS::opencl_constant` is an example of it) are not the same thing which is actually emitted in LLVM IR. There is a map for SPIR target in clang that defines address space numbers https://github.com/intel/llvm/blob/06fdabb5434d43d4e2e4cf4642623db3a7567888/clang/lib/Basic/Targets/SPIR.h#L49

Reply:
May it be requested through some call like getTargetAS(clang::LangAS::opencl_constant)?

Reply:
If you are asking me, I am not aware of such API.

Reply:
I don't think we can use clang APIs in LLVM transformations.
Here is how I w/a'ed similar issue https://github.com/intel/llvm/blob/5e8ad0a35742ca06d33f4919cc23e0f4efb3e447/llvm/lib/SYCLLowerIR/LowerWGScope.cpp#L152

Reply:
Ok, thank you for clarification.

Reply:
I think it is still good idea to use some name instead of '2'